File: niffler-e-2-e-tests/src/test/java/guru/qa/niffler/page/component/SearchField.java
package guru.qa.niffler.page.component;

import com.codeborne.selenide.SelenideElement;

import javax.annotation.Nonnull;
import javax.annotation.ParametersAreNonnullByDefault;

import static com.codeborne.selenide.Condition.empty;
import static com.codeborne.selenide.Condition.not;
import static com.codeborne.selenide.Selenide.$;

@ParametersAreNonnullByDefault
public class SearchField extends BaseComponent<SearchField> {
  public SearchField(SelenideElement self) {
    super(self);
  }

  public SearchField() {
    super($("input[aria-label='search']"));
  }

  private final SelenideElement clearSearchInputBtn = $("#input-clear");

  @Nonnull
  public SearchField search(String query) {
    clearIfNotEmpty();
    self.setValue(query).pressEnter();
    return this;
  }

  @Nonnull
  public SearchField clearIfNotEmpty() {
    if (self.is(not(empty))) {
      clearSearchInputBtn.click();
      self.should(empty);
    }
    return this;
  }
}

File: niffler-e-2-e-tests-python/pages/components/search_field.py
"""SearchField component for search functionality in Niffler application"""

from selene import browser, be, have, command
from selenium.webdriver.common.keys import Keys
import allure

from pages.components.base_component import BaseComponent


class SearchField(BaseComponent):
    """SearchField component matching Java SearchField class"""

    def __init__(self, root=None):
        """
        Initialize search field component

        Args:
            root: Root element of search field. If None, uses default selector
        """
        if root is None:
            root = browser.element('input[aria-label="search"]')
        super().__init__(root)

        self._clear_search_input_btn = browser.element('#input-clear')

    @allure.step("Search for: {query}")
    def search(self, query: str) -> 'SearchField':
        """
        Perform search with given query

        Args:
            query: Search query string

        Returns:
            self for method chaining
        """
        self.clear_if_not_empty()
        self.self.type(query).press_enter()
        return self

    @allure.step("Clear search field if not empty")
    def clear_if_not_empty(self) -> 'SearchField':
        """
        Clear search field if it contains text

        Returns:
            self for method chaining
        """
        # Check if field is not empty
        if self.self.get(be.present).get_attribute('value'):
            self._clear_search_input_btn.click()
            self.self.should(have.value(''))
        return self
